core/trino-spi/src/test/java/io/trino/spi/type/TestDecimals.java

```diff
@@ -162,6 +163,18 @@ public void testEncodeScaledValue()
         assertEquals(encodeScaledValue(new BigDecimal("-172.60"), 2), sliceFromBytes(108, 67, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, minus));
     }
 
+    @Test
+    public void testRescale()
+    {
+        assertEquals(rescale(10233L, 2, 3), 102330);
+        assertEquals(rescale(10253L, 3, 1), 103);
+        assertEquals(rescale(10233L, 3, 2), 1023);
+
+        assertEquals(rescale(new BigInteger("10253"), 2, 3), new BigInteger("102530"));
+        assertEquals(rescale(new BigInteger("10253"), 4, 2), new BigInteger("103"));
+        assertEquals(rescale(new BigInteger("10233"), 4, 2), new BigInteger("102"));
```

Comment:
Add tests to assert an exception is thrown for cases where the value overflows the target type.